packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations.tsx

```diff
@@ -478,18 +478,22 @@ const RelationsInput = ({
      */
     const [targetField] = name.split('.').slice(-1);
 
-    searchForTrigger({
-      model,
-      targetField,
-      params: {
-        ...buildValidParams(query),
-        id: id ?? '',
-        pageSize: 10,
-        idsToInclude: field.value?.disconnect?.map((rel) => rel.id.toString()) ?? [],
-        idsToOmit: field.value?.connect?.map((rel) => rel.id.toString()) ?? [],
-        ...searchParams,
-      },
-    });
+    const timeout = setTimeout(() => {
```

Comment:
Have you checked out the useDebounce hook in the CM?

Reply:
yes but I didn't quite see a way to use it here, and prevent the actual api request. Do you think we could use it here?

Reply:
I would guess something like this pseudocode:

```jsx
const debouncedFieldValue = useDebounce(field.value, 300);

React.useEffect(() => {
  searchForTrigger(debouncedFieldValue);
}, [debouncedFieldValue]);
```

Also I haven't looked into it in detail but I'm not sure why we need a lazy query. A non-lazy one would simplify things:

```
const debouncedFieldValue = useDebounce(field.value, 300);

const { data, isLoading } = useLazySearchRelationsQuery(debouncedFieldValue);
```

Reply:
ah I see! thank you <3

Reply:
that works wonders! great :)

Reply:
I noticed this hook is not documented in the [contributor docs](https://contributor.strapi.io/docs/core/content-manager/hooks/use-content-types). Do you think we should document it now so it is easier to find and use?